pkg/tools/multiexecutor/multi_executor.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package multiexecutor provides serial executor with multiple execution queues
+package multiexecutor
```

Comment:
I'm not sure that this will be used by other chain elements. So can we hide this in `serialize` pkg for a while?

Reply:
Yep.